fdbclient/SpecialKeySpace.actor.cpp

```diff
@@ -1022,6 +1025,18 @@ ACTOR Future<bool> checkExclusion(Database db,
 				if (role["role"].get_str() == "storage") {
 					ssTotalCount++;
 
+					// Check if we are excluding a process that serves the storage role. If this check was true once, we don't have to check any further
+					// since we don't case in this variable about the count of excluded storage servers but only about if we exclude any storage server with the
+					// provided addresses.
+					if !excludedAddressesContainsStorageRole {
+						for (auto exclusion : addresses) {
```

Comment:
I think you can re-use the result of "addressExcluded(*exclusions, addr)" above in place of this "for" loop. This could be as simple as "excludedAddressesContainsStorageRole ||= "the result of addressExcluded(*exclusions, addr) computed above".